hersh/evaluate.test.cpp

```diff
@@ -0,0 +1,9 @@
+#include <hersh/evaluate.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+
+TEST_CASE("Test evaluate", "[evaluate]") {
+    
+    REQUIRE(hersh::evaluate("") == "");
+    REQUIRE(hersh::evaluate("something ") == "something ");
```

Comment:
These should probably separate sections. Otherwise I'm not sure why we need both.

Reply:
I like to always test the empty case.

Reply:
That means you think it's a special case therefore it should have its own section.

Reply:
Partitioning into SECTIONS causes everything between TEST_CASE and the first SECTION to be run before each SECTION. Like a set-up. I don't think there's a need for set-ups here (yet?).

And why should special cases be in a different section even if there's no set-up issues?

Reply:
I'm used to organizing tests in individual test cases and it seems like you're using sections as if they separate test cases. Adding addition assertions (`REQUIRES` in this case) should be done for each test case individually. The test case's name indicates what you're testing.
Therefore having a test case named "check empty case" and one named "check happy path" would be more useful in this case. That way when a test case fails you understand what it's supposed to test immediately by reading its name. I don't know if they should be separate `SECTION`s or separate `TEST_CASE`s but they should be separate something.

Reply:
I think that makes sense. I was following the example in the [Catch2 tutorial](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections) is not a good example?
Are you saying it's not a good example?

Reply:
👍

Reply:
Done. There's no nested TEST_CASE, so you were right, it needs to be partitioned with SECTIONs.